### ISSUE TITLE
Using with webpack 4 fails with error "Module parse failed: Unexpected character '#' (1413:9)"

### ISSUE THREAD

**Issue by Zikoat** [react-pdf-viewer/react-pdf-viewer#1484]
Hi again,

I have a requirement of being on webpack 4, and in that case i am unable to go higher than these versions:

        "@react-pdf-viewer/core": "3.7.0",
        "pdfjs-dist": "^2.6.347",
        "webpack": "^4.46.0",
        
If i push the pdfjs-dist or @react-pdf-viewer core versions one higher, i get this error:

```
ERROR in ./node_modules/pdfjs-dist/build/pdf.js 1413:9
Module parse failed: Unexpected character '#' (1413:9)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|
| class PDFDocumentLoadingTask {
>   static #docId = 0;
|
|   constructor() {
 @ ./node_modules/@react-pdf-viewer/core/lib/cjs/core.js 4:15-36
 @ ./node_modules/@react-pdf-viewer/core/lib/index.js
 @ ./src/App.tsx
 @ ./src/index.tsx
````

Do you have a solution to this error? If not, could you update the docs to tell people that the versions above are the maximum supported versions on webpack 4?

If you need it, i could create a starter project for webpack 4, but the only changes would be these versions and their peer dependencies.

**Comment by phuocng** [comment]
@Zikoat Please take a look at this guide
https://react-pdf-viewer.dev/examples/how-to-solve-the-syntax-error-unexpected-token-issue/

**Comment by deniszatishnyi** [comment]
I have the same for react 17 (create-react-app) and node 16. On node 14 everything was OK

**Comment by deniszatishnyi** [comment]
@phuocng for https://react-pdf-viewer.dev/examples/how-to-solve-the-syntax-error-unexpected-token-issue/ . Did I understand correctly that for `"@react-pdf-viewer/core": "^3.11.0"` and `"pdfjs-dist": "^3.0.279"` I should use the second option (2.8.335 and later)?

**Comment by deniszatishnyi** [comment]
@phuocng Did I understand correctly that in the latest commits, peerDependencies are changed from

```
    "peerDependencies": {
        "pdfjs-dist": "^2.16.105 || ^3.0.279",
        "react": ">=16.8.0",
        "react-dom": ">=16.8.0"
    },
```

to

```
    "peerDependencies": {
        "react": ">=16.8.0",
        "react-dom": ">=16.8.0"
    },
```
?

If so, this is not reflected in the latest [release 3.12.0](https://github.com/react-pdf-viewer/react-pdf-viewer/blob/v3.12.0/packages/core/package.json) and I still get an error on node 16, which requires strict adherence to peerDependencies when I try to use `"pdfjs-dist": "2.6.347"`

**Comment by Zikoat** [comment]
@phuocng Yes, the guide you sent worked very well! The guide is only linked to from the "troubleshooting" page, and not from other places in the documentation, which is why i probably overlooked it.

Using the guide i was able to upgrade my dependencies to:

```
    "@react-pdf-viewer/core": "3.12.0",
    "pdfjs-dist": "2.16.105",
```

However, when trying to upgrade pdfjs-dist to version 3.0.279 or 3.5.141, i get the following error:

```
 error  in ./node_modules/pdfjs-dist/legacy/build/pdf.js

Module parse failed: Unexpected token (5936:137)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
|       numPages: this._numPages,
|       annotationStorage: this.annotationStorage.serializable,
>       filename: ((_this$_fullReader = this._fullReader) === null || _this$_fullReader === void 0 ? void 0 : _this$_fullReader.filename) ?? null
|     }).finally(() => {
|       this.annotationStorage.resetModified();

 @ ./node_modules/@react-pdf-viewer/core/lib/cjs/core.js 4:15-36
 @ ./node_modules/@react-pdf-viewer/core/lib/index.js
```

I tried with both of these options in webpack config, but they show the same error.

```
module.exports = {
  resolve: {
    extensions: ['.js', '.jsx', '.ts', '.tsx'],
    alias: {
      // 'pdfjs-dist': path.resolve('./node_modules/pdfjs-dist/es5/build/pdf.js'),
      'pdfjs-dist': path.resolve('./node_modules/pdfjs-dist/legacy/build/pdf.js'),
    },
  },
```

Have you successfully used pdfjs-dist 3.x.x with webpack 4? Do you need a repository with a reproduction of this issue, or is this a known issue?

**Comment by Zikoat** [comment]
@deniszatishnyi No, `@react-pdf-viewer/core@3.12.0` still has this line in it's peer dependencies:
```
        "pdfjs-dist": "^2.16.105 || ^3.0.279",
```

Which means that `pdfjs-dist@2.6.347` is not a valid peer dependency, but `pdfjs-dist@2.16.105` is valid.

If you are getting the error i have above, you will have to upgrade pdfjs-dist to 2.16.105 and follow the instructions in this guide: https://react-pdf-viewer.dev/examples/how-to-solve-the-syntax-error-unexpected-token-issue/

**Comment by deniszatishnyi** [comment]
@Zikoat thank you. But when I use `"pdfjs-dist": "2.16.105"` instead of `"pdfjs-dist": "^2.16.105"` on node 16 and react 17 (using CRA) I still get the error. On the node 16 it works fine

**Comment by Zikoat** [comment]
@deniszatishnyi "pdfjs-dist": "2.16.105" and "pdfjs-dist": "^2.16.105" should be the same, so i don't know why one of them works and one doesn't.

Do you get the error `Unexpected character '#'`? Have you followed the guide? If you are still getting the error, could you create a repro?

**Comment by deniszatishnyi** [comment]
@Zikoat 

> Latest CRA also [seems to use webpack 5,](https://github.com/facebook/create-react-app/blob/d960b9e38c062584ff6cfb1a70e1512509a966e7/packages/react-scripts/package.json#L74) are you sure you are using webpack 4?

right, webpack 5

**Comment by Zikoat** [comment]
Remaining question is the one posted in the previous question:

> Have you successfully used pdfjs-dist 3.x.x with webpack 4? Do you need a repository with a reproduction of this issue, or is this a known issue?

If you have tried and not successfully used these versions together then this issue can be closed, but if you have some steps i can take to use pdfjs-dist 3 with webpack 4, then that would be awesome.
